### DISCUSSION TITLE
Pack and cell voltages using 2RC Thevenin Model

### DISCUSSION THREAD

**Discussion by MustafaDragonborne** [pybamm-team/liionpack#253] [Q&A]
When simulating a 1p100s battery pack using a 2RC thevenin model, I noticed that the output pack voltage creates very different values than what I would expect. It does not show the kind of dynamic behaviour that I would expect the battery pack to show when resting. Moreoever, the pack terminal voltage shows an abnormal spike during the transition from rest to discharge and the comes back down.

The following figures were generated by running an experiment as follows:
1. Charge at 40A for 100 seconds
2. Rest for 100 seconds
3. Discharge at 40A for 100 seconds

The sum of the individual cell voltages show the behaviour that I would expect but the pack voltages do not. This is also a potential bug in the way V_terminal is computed inside the model.
![image](https://user-images.githubusercontent.com/64160706/230406520-0c3009ab-3699-4f2d-bcf5-1153c7673efe.png)

Moreover, the cell voltages do not seem to be perfect either. A zoom in at the transition point where the battery goes from charging to rest shows that the instance the current goes to 0, the cell voltage does not jump down but rather stays at the same value and only goes down at the next time step. A similar effect can be seen at the transition point where the battery goes from rest to discharge.

![image](https://user-images.githubusercontent.com/64160706/230412534-2ef10eb6-033d-419f-8057-555b89f2854a.png)
![image](https://user-images.githubusercontent.com/64160706/230412628-490ae57f-8cb7-4015-b992-ef234f85f809.png)

Code to reproduce the issue:
```
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

import sys
sys.path.append('/home/mustafa/Documents/ECM_parameter_sets')
import liTec_Prototyp_40Ah_Zelle1
import time
from datetime import timedelta

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib tk
from sklearn.metrics import mean_absolute_error, mean_squared_error

import pybamm
from pybamm import ParameterValues
pybamm.set_logging_level('NOTICE')
import liionpack as lp

# set the parameter values according to chemistry
parameter_values_litec = liTec_Prototyp_40Ah_Zelle1.get_parameter_values()
# remove unwanted keys
unwanted_keys = 'chemistry'
del parameter_values_litec[unwanted_keys]
parameter_values_litec = ParameterValues(parameter_values_litec)
mean_start_SoC = 0.5
stddev_start_SoC = 0.01
Np = 1  # number of cells in parallel
Ns = 100  # number of cells in series
init_SoC_dist = np.random.normal(loc=mean_start_SoC, scale=stddev_start_SoC, size=Np*Ns)
parameter_values_litec.update({'Initial SoC': '[input]'})
inputs = {'Initial SoC': init_SoC_dist}  # gaussian spread

experiment = pybamm.Experiment(
    [
        "Charge with 40A for 100 seconds",
        "Rest for 100 seconds",
        "Discharge with 40A for 100 seconds",        
    ] * 1,
    period='0.5 second'
)
def mysim_func(parameter_values):
    # create the model
    model = pybamm.equivalent_circuit.Thevenin(options={'number of rc elements':2})

    # Set up solver and simulation
    solver = pybamm.CasadiSolver(mode="fast")
    sim = pybamm.Simulation(
        model=model,
        parameter_values=parameter_values,
        solver=solver
    )
    return sim

netlist = lp.setup_circuit(
    Np=Np, Ns=Ns, Ri=1e-10, Rc=1e-10, Rb=1e-10, Rt=1e-10
)

t0 = time.time()
output = lp.solve(
    netlist=netlist,
    sim_func=mysim_func,
    parameter_values=parameter_values_litec,
    experiment=experiment,
    inputs=inputs,
)
t1 = time.time()
```

**Comment by TomTranter** [comment]
This problem is not reproducible as we don't have access to your parameter set. Can you provide a fully working example

**Reply by MustafaDragonborne** [reply]
Hi Tom, 
apologies for the late reply. Please save the contents of the zip file in a folder and replace the path of the folder within line 4 of sys.path.append with your path. Also do not forget to include an `__init__.py` file inside the folder to be able to import it. Please let me know if it works or not. 
[liTec_Prototyp_40Ah_Zelle1.zip](https://github.com/pybamm-team/liionpack/files/11198163/liTec_Prototyp_40Ah_Zelle1.zip)

**Reply by MustafaDragonborne** [reply]
Moreover, the 1s forward displacement of the cell voltages looks similar to the problem in #242

**Reply by MustafaDragonborne** [reply]
@TomTranter I ran a similar experiment using the default electrochemical model and the Chen2020 parameter set and the results were as expected. The pack terminal voltage and the cell voltages showed the dynamic behaviour as expected. So this is a problem that is also very ECM specific. 

EDIT: Please ignore some parts of the titles as they can be misleading (like drive cycle duration) since I forgot to change them from my last simulations.

![image](https://user-images.githubusercontent.com/64160706/231437290-bc55a4ca-0fa4-414a-a482-d332da36b9d8.png)

Here's the code:

```
import time
from datetime import timedelta
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib tk
from sklearn.metrics import mean_absolute_error, mean_squared_error
import pybamm
from pybamm import ParameterValues
pybamm.set_logging_level('NOTICE')
import liionpack as lp

# set the parameter values according to chemistry
parameter_values_chen = pybamm.ParameterValues('Chen2020')
Np = 1  # number of cells in parallel
Ns = 100  # number of cells in series

experiment = pybamm.Experiment(
    [
        "Charge with 5A for 100 seconds",
        "Rest for 100 seconds",
        "Discharge with 5A for 100 seconds",        
    ] * 1,
    period='0.5 second'
)
netlist = lp.setup_circuit(
    Np=Np, Ns=Ns, Ri=1e-10, Rc=1e-10, Rb=1e-10, Rt=1e-10
)

t0 = time.time()
output = lp.solve(
    netlist=netlist,
    parameter_values=parameter_values_chen,
    experiment=experiment,
    initial_soc=0.5
)
t1 = time.time()
```

**Comment by sjjigh92** [comment]
How can I see and change R0, R1,R2, C1, C2, in Thevenin ECM, here?
